src/nile/core/deploy.py

```diff
@@ -39,3 +54,52 @@ def deploy(
 
     deployments.register(address, register_abi, network, alias)
     return address, register_abi
+
+
+def deploy_contract(
+    account,
+    contract_name,
+    salt,
+    unique,
+    calldata,
+    alias,
+    deployer_address,
+    max_fee,
+    overriding_path=None,
+    abi=None,
+):
+    """Deploy StarkNet smart contracts through UDC."""
+    logging.info(f"🚀 Deploying {contract_name}")
+
+    base_path = (
+        overriding_path if overriding_path else (BUILD_DIRECTORY, ABIS_DIRECTORY)
+    )
+    register_abi = abi if abi is not None else f"{base_path[1]}/{contract_name}.json"
+    deployer_for_address_generation = 0
+
+    if unique:
+        # Match UDC address generation
+        salt = compute_hash_chain(data=[account.address, salt])
+        deployer_for_address_generation = deployer_address
```

Comment:
What do you think about including a random salt generator for deployments without `--salt` like StarkWare does? => https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/cli/starknet_cli.py#L846-L847

Otherwise, trying to deploy multiple instances of the same contract will result in:

`raise StarkException(code=code, message=message)
starkware.starkware_utils.error_handling.StarkException: (500, {'code': <StarknetErrorCode.CONTRACT_ADDRESS_UNAVAILABLE: 1>, 'message': 'Requested contract address 1999693730423262048654525999365022023777068478432229193013554800567845966099 is unavailable for deployment.'})`

Users _can_ add `salt` to avoid this, but automating salt seems preferable

Reply:
If we agree^, I definitely suggest adding some salt tests

Reply:
I think this salt should be part of the protocol (like nonce in Ethereum deployments), but I am not sure why it is left to be passed from the end users (am I missing an important reason?). With this said, agreeing on a format for the automatic generation of this salt, I think, is content enough for another issue/PR. Shouldn't be any problem with releasing this version, right? I don't see a reason to delay it to another milestone.

Reply:
Fixed.

Reply:
> I think this salt should be part of the protocol

I agree in general haha but since we're deploying contracts through an `account.send` tx, the burden seems to fall on Nile. When Nile interacts with the starknet_cli directly (and when Nile most likely bypasses the cli in the future), I imagine defining a salt generator will not be necessary

> agreeing on a format for the automatic generation of this salt, I think, is content enough for another issue/PR

That sounds reasonable to me

> I don't see a reason to delay it to another milestone.

I don't disagree